PLP/PLE export error

### Expected behavior
Export works.

### Actual behavior
Export doesn't work for updated PLP/PLE.

### Steps to reproduce behavior
- create and save correct PLP or PLE
- open it and go to "Utilities" tab - export shoud be ok
- change something (a name, for example), save
- open and go to "Utilities" tab - export fails with 500 error code

Meantime, on the backend:
```
2020-08-17 13:03:13.381 ERROR http-nio-8080-exec-7 org.ohdsi.webapi.util.GenericExceptionMapper -  - java.lang.reflect.UndeclaredThrowableException
	at com.sun.proxy.$Proxy226.exportAnalysis(Unknown Source)
	at org.ohdsi.webapi.estimation.EstimationController.exportAnalysis(EstimationController.java:154)
<...>
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.lang.String out of START_OBJECT token
 at <...> (through reference chain: org.ohdsi.webapi.estimation.specification.EstimationAnalysisImpl["createdBy"])
	at org.ohdsi.webapi.estimation.EstimationServiceImpl.exportAnalysis(EstimationServiceImpl.java:223)
```

Cause of the error is the field "createdBy" in EstimationAnalysisImpl. It is String. After the first save, "createdBy" is null and not causes problems while exporting. After second save it changes to object (i.e. "createdBy":{"id":1000,"name":"admin@odysseusinc.com","login":"admin@odysseusinc.com"}) and cannot be deserialized as String.

Looks like PLP/PLE specification for the second save (update) contains incorrect createdBy.

--- a/js/pages/estimation/cca-manager.js
+++ b/js/pages/estimation/cca-manager.js
@@ -192,7 +192,7 @@ define([
 			GlobalPermissionService.decorateComponent(this, {
 				entityTypeGetter: () => entityType.ESTIMATION,
 				entityIdGetter: () => this.selectedAnalysisId(),
-				createdByUsernameGetter: () => this.estimationAnalysis() && lodash.get(this.estimationAnalysis(), 'createdBy.login')
+				createdByUsernameGetter: () => this.estimationAnalysis() && lodash.get(this.estimationAnalysis(), 'createdBy')
 			});
 		}
 

--- a/js/pages/estimation/cca-manager.js
+++ b/js/pages/estimation/cca-manager.js
@@ -383,7 +383,12 @@ define([
 			const specification = JSON.parse(analysis.data.specification);
 			// ignore createdBy and modifiedBy
 			const { createdBy, modifiedBy, ...props } = header;
-			this.estimationAnalysis(new EstimationAnalysis({ ...specification, ...header }, this.estimationType, this.defaultCovariateSettings()));
+			this.estimationAnalysis(new EstimationAnalysis({
+				...specification,
+				...props,
+				createdBy: createdBy ? createdBy.name : null,
+				modifiedBy: modifiedBy ? modifiedBy.name : null
+			}, this.estimationType, this.defaultCovariateSettings()));
 			this.estimationAnalysis().id(header.id);
 			this.estimationAnalysis().name(header.name);
 			this.estimationAnalysis().description(header.description);
@@ -544,13 +549,13 @@ define([
 			const createdDate = commonUtils.formatDateForAuthorship(this.estimationAnalysis().createdDate);
 			const modifiedDate = commonUtils.formatDateForAuthorship(this.estimationAnalysis().modifiedDate);
 			return {
-					createdBy: lodash.get(this.estimationAnalysis(), 'createdBy.name'),
+					createdBy: lodash.get(this.estimationAnalysis(), 'createdBy'),
 					createdDate,
-					modifiedBy: lodash.get(this.estimationAnalysis(), 'modifiedBy.name'),
+					modifiedBy: lodash.get(this.estimationAnalysis(), 'modifiedBy'),
 					modifiedDate,
 			}
 		}
 	}
 
-	return commonUtils.build('cca-manager', ComparativeCohortAnalysisManager, view);;
+	return commonUtils.build('cca-manager', ComparativeCohortAnalysisManager, view);
 });

--- a/js/pages/prediction/prediction-manager.js
+++ b/js/pages/prediction/prediction-manager.js
@@ -400,7 +400,13 @@ define([
 		loadAnalysisFromServer(analysis) {
 			var header = analysis.json;
 			var specification = JSON.parse(analysis.data.specification);
-			this.patientLevelPredictionAnalysis(new PatientLevelPredictionAnalysis({ ...specification, ...header }));
+			const { createdBy, modifiedBy, ...props } = header;
+			this.patientLevelPredictionAnalysis(new PatientLevelPredictionAnalysis({
+				...specification,
+				...props,
+				createdBy: createdBy ? createdBy.name : null,
+				modifiedBy: modifiedBy ? modifiedBy.name : null
+			}));
 			this.packageName(header.packageName);
 			this.setUserInterfaceDependencies();
 			this.setAnalysisSettingsLists();
@@ -456,9 +462,9 @@ define([
 			const createdDate = commonUtils.formatDateForAuthorship(this.patientLevelPredictionAnalysis().createdDate);
 			const modifiedDate = commonUtils.formatDateForAuthorship(this.patientLevelPredictionAnalysis().modifiedDate);
 			return {
-					createdBy: lodash.get(this.patientLevelPredictionAnalysis(), 'createdBy.name'),
+					createdBy: lodash.get(this.patientLevelPredictionAnalysis(), 'createdBy'),
 					createdDate,
-					modifiedBy: lodash.get(this.patientLevelPredictionAnalysis(), 'modifiedBy.name'),
+					modifiedBy: lodash.get(this.patientLevelPredictionAnalysis(), 'modifiedBy'),
 					modifiedDate,
 			}
 		}

--- a/js/pages/prediction/prediction-manager.js
+++ b/js/pages/prediction/prediction-manager.js
@@ -195,7 +195,7 @@ define([
 			GlobalPermissionService.decorateComponent(this, {
 				entityTypeGetter: () => entityType.PREDICTION,
 				entityIdGetter: () => this.selectedAnalysisId(),
-				createdByUsernameGetter: () => this.patientLevelPredictionAnalysis() && lodash.get(this.patientLevelPredictionAnalysis(), 'createdBy.login')
+				createdByUsernameGetter: () => this.patientLevelPredictionAnalysis() && lodash.get(this.patientLevelPredictionAnalysis(), 'createdBy')
 			});
 		}
 

--- a/js/services/Permission.js
+++ b/js/services/Permission.js
@@ -41,7 +41,7 @@ define(function (require, exports) {
 			return createdByUsernameGetter() === username;
 		};
 
-		component.isOwner = ko.computed(() => component.isOwnerFn(authApi.subject()));
+		component.isOwner = config.userAuthenticationEnabled && ko.computed(() => component.isOwnerFn(authApi.subject()));
 
 		component.loadAccessList = () => {
 			return loadEntityAccessList(entityTypeGetter(), entityIdGetter());

